server/authentication/auth.ts

```diff
@@ -0,0 +1,63 @@
+import passport from 'passport'
+import { Strategy } from 'passport-oauth2'
+import type { Request, RequestHandler } from 'express'
+
+import config from '../config'
+import generateOauthClientToken from './clientCredentials'
+import type { VerifiableRequest, TokenVerifier } from '../data/tokenVerification'
+
+passport.serializeUser((user, done) => {
+  // Not used but required for Passport
+  done(null, user)
+})
+
+passport.deserializeUser((user, done) => {
+  // Not used but required for Passport
+  done(null, user)
+})
+
+export type AuthenticationMiddleware = (tokenVerifier: TokenVerifier) => RequestHandler
+
+const authenticationMiddleware: AuthenticationMiddleware = verifyToken => {
+  return async (req: VerifiableRequest, res, next) => {
+    if (req.isAuthenticated() && (await verifyToken(req))) {
+      return next()
+    }
+    req.session.returnTo = req.originalUrl
+    return res.redirect('/login')
+  }
+}
+
+export interface UserDetails {
+  username: string
+  token: string
+  authSource: string
+}
+
+export interface UserRequest extends Request {
```

Comment:
Can't quite see how this is used.

In use of force we override Express request definition globally to add this - there are a couple of issues with it some would be good if this is a better solution